src/lib/FreeC/IR/Subterm.hs

```diff
@@ -247,6 +274,10 @@ replaceSubterms term ((p, e) : pes) = do
   term' <- replaceSubterm term p e
   replaceSubterms term' pes
 
+-- | Like 'replaceSubterms' but throws an error if any of the subterms could
+--   not be replaced.
+replaceSubterms' :: Subterm a => a -> [(Pos, a)] -> a
+replaceSubterms' = foldl (\term (pos, term') -> replaceSubterm' term pos term')
```

Comment:
I like the solution using `foldl`. Could you rewrite `replaceSubterms` using `foldlM` as well? : )